src/ducks/Signals/SignalMaster.js

```diff
@@ -37,19 +37,38 @@ export class SignalMaster extends React.PureComponent {
     }
     const { step, trigger } = this.state
     const currentTrigger = trigger || (this.props.trigger || {}).trigger
-    const formProps = currentTrigger
+    let formProps = currentTrigger
       ? mapTriggerToFormProps(currentTrigger)
       : undefined
     const meta = {
       title: currentTrigger ? currentTrigger.title : 'Any',
       description: currentTrigger ? currentTrigger.description : 'Any'
     }
+
+    let metaFormSettings = { ...this.props.metaFormSettings }
+
+    if (!this.props.metaFormSettings && this.props.asset) {
+      metaFormSettings = {
+        ...metaFormSettings,
+        ...{
+          target: {
+            isDisabled: false,
+            value: {
+              value: this.props.asset,
+              label: this.props.asset
```

Comment:
1. `value` object has a `value` property. That's not really great for the code readability/semantics;
2. `value` and `label` property has the same value `this.props.asset`, so the question is "why not to have the single property?";
3. Same property access in multiple places inflates the code and influence the performance. Destructuring assignment should be used in the current case

Reply:
Copypaste, will fix